File: ProblemSet6 - Python/vigenere/vigenere.py
import cs50
import sys

def main():
    if len(sys.argv) <= 1 or len(sys.argv) > 2:
        exit(1)

    if key_check(sys.argv[1]) == 0:
        exit(1)

    print("plaintext: ", end="")
    plaintext = cs50.get_string()
    print("ciphertext: ", end="")

    m = 0

    for i in range(len(plaintext)):
        if plaintext[i] != ' ' and str.isalpha(plaintext[i]):
            k = key(sys.argv[1], m)
            m += 1

            if str.isupper(plaintext[i]):
                print("{}".format(crypt_upper(plaintext[i], k)), end="")
            elif str.islower(plaintext[i]):
                print("{}".format(crypt_lower(plaintext[i], k)), end="")
        else:
            print("{}".format(plaintext[i]), end="")

    print()

def key_check(key):
    for i in range(len(key)):
        if not str.isalpha(key[i]):
            return 0

    return 1

def key(key, number):

    key_num = number % len(key)
    alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    k = 0

    key2 = key.upper()
    #print("{}".format(key))

    for i in range(26):
        if alpha[i] == key2[key_num]:
            k = i

    return k

def crypt_upper(c, k):
    alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"

    for i in range(26):
        if c == alpha[i]:
            crypt_char = alpha[(i+k) % 26]

    return crypt_char

def crypt_lower(c, k):
    alpha = "abcdefghijklmnopqrstuvwxyz"

    for i in range(26):
        if c == alpha[i]:
            crypt_char = alpha[(i+k) % 26]

    return crypt_char

if __name__ == "__main__":
    main()

File: ProblemSet2-Strings,Arrays/vigenere.c
// modified caesar encryption. enter keyword as a command line argument (only alphabetic letters)

#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

#define ALPHA_CHAR 26

int key(string key, int i);
char crypt_upper(char c, int k);
char crypt_lower(char c, int k);
_Bool key_check(string key);

int main(int argc, string argv[]){

    int k =0;
    if(argc<=1 || argc > 2){
        return 1;
    }

    if(key_check(argv[1]) == 0){
        return 1;
    }

    printf("plaintext: ");
    string plaintext = get_string();
    printf("ciphertext: ");

    int m = 0;

    for(int i = 0, n = strlen(plaintext); i < n; i++){
        if(plaintext[i] != ' ' && isalpha(plaintext[i])){
            k = key(argv[1], m);
            m++;

            if(isupper(plaintext[i])){
                printf("%c", crypt_upper(plaintext[i], k));
            }
            else if(islower(plaintext[i])){
                printf("%c", crypt_lower(plaintext[i], k));
            }
        }
        else{
            printf("%c", plaintext[i]);
        }
    }
    printf("\n");
    return 0;
}

_Bool key_check(string key){

    for(int i =0, n = strlen(key); i<n; i++){
        if(!isalpha(key[i])){
            return 0;
        }
    }
    return 1;
}

int key(string key, int number){

    int key_num = number % strlen(key); //0
    string alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    int k = 0;

    for(int i =0, n = strlen(key); i<n; i++){
        key[i] = toupper(key[i]);
    }

    for(int i = 0; i < ALPHA_CHAR; i++){
        if(alpha[i] == key[key_num])
            k = i;
    }
    return k;
}

char crypt_upper(char c, int k){
    string alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    char crypt_char;

    for(int i =0; i<ALPHA_CHAR; i++){
        if(c == alpha[i])
            crypt_char = alpha[(i+k)%26];
    }
    return crypt_char;
}

char crypt_lower(char c, int k){
    string alpha = "abcdefghijklmnopqrstuvwxyz";
    char crypt_char;

    for(int i =0; i < ALPHA_CHAR; i++){
        if(c == alpha[i])
            crypt_char = alpha[(i+k)%26];
    }
    return crypt_char;
}
